cloudify_gcp/compute/instance.py

```diff
@@ -249,26 +274,27 @@ def list(self):
 
         return self.discovery.instances().list(
             project=self.project,
-            zone=self.zone).execute()
+            zone=basename(self.zone)).execute()
 
     def to_dict(self):
         def add_key_value_to_metadata(key, value, body):
             body['metadata']['items'].append({'key': key, 'value': value})
 
-        def get_instance_ssh_keys():
-            agent_key = utils.get_agent_ssh_key_string()
-            other_keys = ctx.instance.runtime_properties.get(constants.SSHKEY)
-            return other_keys + '\n' + agent_key if other_keys else agent_key
+        network = {'network': 'global/networks/default'}
+        if self.network and self.network != 'default':
+            network['network'] = self.network
+        if self.subnetwork:
+            network['subnetwork'] = self.subnetwork
```

Comment:
External ip doesn't work. It was not tested for a long time so maybe api has changed. Error:  File "/root/b/env/lib/python2.7/site-packages/cloudify_gcp/compute/instance.py", line 328, in to_dict
    if item['name'] == self.ACCESS_CONFIG:
KeyError: 'name'